app/mailers/user_mailer.rb

```diff
@@ -14,71 +17,87 @@ class UserMailer < ActionMailer::Base
     ),
   )
 
-  def email_confirmation_instructions(user, email, token, request_id:, instructions:)
-    with_user_locale(user) do
-      presenter = ConfirmationEmailPresenter.new(user, view_context)
+  def validate_user_and_email_address
+    @user = params.fetch(:user)
+    @email_address = params.fetch(:email_address)
+    if @user.id != @email_address.user_id
+      raise UserEmailAddressMisMatchError.new(
+        "User ID #{@user.id} does not match EmailAddress ID #{@email_address.id}",
+      )
+    end
+  end
```

Comment:
do you think it would make sense to just require an `EmailAddress` record and not ever have a `user:`?

Reply:
I was kind of waffling on that since it's a good simplification. I lean towards keeping both for now for the explicitness and avoiding a potentially duplicative query. If we move to async, that is kind of moot though, so I'd be cool with revisiting?

Reply:
+1 revisiting later seems fine